pkg/codegen/python/gen.go

```diff
@@ -867,8 +865,19 @@ func (mod *modContext) importResourceType(r *schema.ResourceType) string {
 		name = "provider"
 	}
 
-	components := strings.Split(modName, "/")
-	return fmt.Sprintf("from %s%s import %s", importPath, name, components[0])
+	modName := mod.tokenToModule(tok)
+	importPath := mod.getRelImportFromRoot(modName)
+	if importPath == "." {
+		importPath = "." + name
+	} else {
+		importPath = importPath + "." + name
+	}
+
+	resourceName := mod.tokenToResource(tok)
+	components := strings.Split(resourceName, ".")
```

Comment:
I think this part was always broken. It was splitting on `/`, but mod.tokenToResource(tok) always replaces `/` with `.` dating back to when this was first introduced in https://github.com/pulumi/pulumi/commit/81580289a74c6bb13b4f238f297a48c3c2a8b16b#diff-f96a068940ee11b51a8bb4e6bbd2289ff1d18dc300bb90f527868eee0a0097f1R164